OSDOCS-3969 Alibaba VPC installation - new assembly

GroceryBoyJr opened the pull request:
Status: **Peer Review Requested. I have SME LGTM, waiting for QE LGTM.**

Problem:  This PR adds a new section documenting Alibaba VPC installation

Version(s): 4.10+

Issue:
How to install OpenShift into an Alibaba VPC  https://issues.redhat.com/browse/OSDOCS-3969

Link to docs preview:
http://file.rdu.redhat.com/jbrigman/alibaba-vpc/installing/installing_alibaba/installing-alibaba-vpc.html

Additional information:
Gaurav Singh, PM. QE: Jianli Wei, SME: Thomas Wiest (Thomas has left the company but provided LGTM before departure and as-of the last QE update from Jianli Wei.)

jianli-wei commented on installing/installing_alibaba/installing-alibaba-vpc.adoc:
```diff
@@ -0,0 +1,71 @@
+* You xref:../../installing/installing_alibaba/preparing-to-install-on-alibaba.adoc#installation-alibaba-dns_preparing-to-install-on-alibaba[registered your domain].
+* If you use a firewall, you xref:../../installing/install_config/configuring-firewall.adoc#configuring-firewall[configured it to allow the sites] that your cluster requires access to.
+* If the cloud Resource Access Management (RAM) APIs are not accessible in your environment, or if you do not want to store an administrator-level credential secret in the `kube-system` namespace, you can xref:../../installing/installing_alibaba/manually-creating-alibaba-ram.adoc#manually-creating-alibaba-ram[manually create and maintain Resource Access Management (RAM) credentials].
+
```
Suggest to add section `About using a custom VPC` and section `Requirements for using your VPC` (For example, please refer to [Installing a cluster on GCP into an existing VPC](https://docs.openshift.com/container-platform/4.10/installing/installing_gcp/installing-gcp-vpc.html)). 

When using an existing VPC, the installation program does not create the following components: 
- VPC
- vSwitches
- Route table
- Internet NAT Gateway

GroceryBoyJr replied:
Agreed! @jianli-wei I'm adding those sections and a little more for you to review, please.

jianli-wei commented on installing/installing_alibaba/installing-alibaba-vpc.adoc:
```diff
@@ -0,0 +1,71 @@
+
+include::modules/cco-ccoctl-creating-at-once.adoc[leveloffset=+2]
+
+include::modules/installation-configuration-parameters.adoc[leveloffset=+2]
```
Suggest to move the section to rightly after the section `Creating the installation configuration file`.

GroceryBoyJr replied:
Done! Thank you @jianli-wei and @twiest ! Great catch!

jianli-wei commented on installing/installing_alibaba/installing-alibaba-vpc.adoc:
```diff
@@ -0,0 +1,71 @@
+
+include::modules/installation-configuration-parameters.adoc[leveloffset=+2]
+
+include::modules/installation-alibaba-config-yaml.adoc[leveloffset=+2]
```
Suggest to tell a sample section for specifying vpcID, for example,

platform:
  alibabacloud:
    region: us-east-1
    vpcID: vpc-0xifdjerdibmaqvtjob2b
    vswitchIDs:
    - vsw-0xi8ycgwc8wv5rhviwdq5
    - vsw-0xiy6v3z2tedv009b4pz2

jianli-wei replied:
@GroceryBoyJr It seems the current doc preview doesn't have the `vpcID` and `vswitchIDs` sections, suggest to add, thanks!

jianli-wei commented on installing/installing_alibaba/installing-alibaba-vpc.adoc:
```diff
@@ -0,0 +1,71 @@
+
+include::modules/installation-alibaba-config-yaml.adoc[leveloffset=+2]
+
+include::modules/installation-configure-proxy.adoc[leveloffset=+2]
```
Suggest to remove the section, because it's not tested on Alibabacloud (due to bug https://bugzilla.redhat.com/show_bug.cgi?id=2046226).

jianli-wei replied:
@GroceryBoyJr Any updates? Thanks!

jianli-wei commented:
@GroceryBoyJr Another 2 suggestions, thanks! 

Replace "alibaba" with "alibabacloud" in below parameters' description.
Table 3. Optional parameters
  compute.platform
  controlPlane.platform

Remove the parameter from the optional params list. 
Table 4. Optional Alibaba Cloud parameters
  platform.alibabacloud.region

GroceryBoyJr commented:
> > Problem: This PR adds a new section that works correctly for Alibaba VPC installation
> > Version(s): 4.8+
> > Issue: https://bugzilla.redhat.com/show_bug.cgi?id=
> > Link to docs preview: http://file.rdu.redhat.com/jbrigman/alibaba-vpc/installing/installing_alibaba/installing-alibaba-vpc.html
> > Additional information: Initial review needed by product owner Gaurav Singh
> 
> The `Version(s)` should be 4.10+, thanks!

Changed, thank  you!

GroceryBoyJr commented:
> @GroceryBoyJr Another 2 suggestions, thanks!
> 
> Replace "alibaba" with "alibabacloud" in below parameters' description. Table 3. Optional parameters compute.platform controlPlane.platform
> 
> Remove the parameter from the optional params list. Table 4. Optional Alibaba Cloud parameters platform.alibabacloud.region

This is done, thank you so much @jianli-wei

jianli-wei commented:
Hi @GroceryBoyJr, some other final suggestions, if possible, please address, thanks in advance!

1. Below parameters are for both compute machines and control plane machines, if not otherwise specified.
_Table 4. Optional Alibaba Cloud parameters_

- platform.alibabacloud.defaultMachinePlatform.imageID (_btw, it's description statement missing the ending dot_)
- platform.alibabacloud.defaultMachinePlatform.instanceType
- platform.alibabacloud.defaultMachinePlatform.systemDiskCategory
- platform.alibabacloud.defaultMachinePlatform.systemDiskSize
- platform.alibabacloud.defaultMachinePlatform.zones

2. Table 4 doesn't list parameters under compute.platform.alibabacloud and controlPlane.platform.alibabacloud, which would overwrite platform.alibabacloud.defaultMachinePlatform for compute machines and control plane machines respectively. Suggest to add a statement telling it before the table.

3. Suggest to move [Sample customized install-config.yaml file for Alibaba Cloud](http://file.rdu.redhat.com/jbrigman/alibaba-vpc/installing/installing_alibaba/installing-alibaba-vpc.html#installation-alibaba-config-yaml_installing-alibaba-vpc) to before [Generating the required installation manifests](http://file.rdu.redhat.com/jbrigman/alibaba-vpc/installing/installing_alibaba/installing-alibaba-vpc.html#manually-creating-alibaba-manifests_installing-alibaba-vpc).

GroceryBoyJr commented:
> > Problem: This PR adds a new section that works correctly for Alibaba VPC installation
> > Version(s): 4.8+
> > Issue: https://bugzilla.redhat.com/show_bug.cgi?id=
> > Link to docs preview: http://file.rdu.redhat.com/jbrigman/alibaba-vpc/installing/installing_alibaba/installing-alibaba-vpc.html
> > Additional information: Initial review needed by product owner Gaurav Singh
> 
> The `Version(s)` should be 4.10+, thanks!

Done, thank you!

GroceryBoyJr commented:
> Hi @GroceryBoyJr, some other final suggestions, if possible, please address, thanks in advance!
> 
> 1. Below parameters are for both compute machines and control plane machines, if not otherwise specified.
>    _Table 4. Optional Alibaba Cloud parameters_
> 
> * platform.alibabacloud.defaultMachinePlatform.imageID (_btw, it's description statement missing the ending dot_)
> * platform.alibabacloud.defaultMachinePlatform.instanceType
> * platform.alibabacloud.defaultMachinePlatform.systemDiskCategory
> * platform.alibabacloud.defaultMachinePlatform.systemDiskSize
> * platform.alibabacloud.defaultMachinePlatform.zones
> 
Added the note to the text just above the table (and fixed the missing dot, thank you!)

> 2. Table 4 doesn't list parameters under `compute.platform.alibabacloud` and `controlPlane.platform.alibabacloud`, which would overwrite `platform.alibabacloud.defaultMachinePlatform` for compute machines and control plane machines respectively. Suggest to add a statement telling it before the table.

Added a NOTE saying this, just above the table.

> 3. Suggest to move [Sample customized install-config.yaml file for Alibaba Cloud](http://file.rdu.redhat.com/jbrigman/alibaba-vpc/installing/installing_alibaba/installing-alibaba-vpc.html#installation-alibaba-config-yaml_installing-alibaba-vpc) to before [Generating the required installation manifests](http://file.rdu.redhat.com/jbrigman/alibaba-vpc/installing/installing_alibaba/installing-alibaba-vpc.html#manually-creating-alibaba-manifests_installing-alibaba-vpc).

Moved example YAML section to where you requested.

Thank you gents!

GroceryBoyJr commented:
@jianli-wei @twiest All requested changes are complete - PTAL.

jianli-wei commented:
@GroceryBoyJr Sorry, it seems I didn't put it very clear on the table 4 suggestions, let me clarify. Very sorry to bother you again. 

1. Now in table 4, e.g. the Description field of `platform.alibabacloud.defaultMachinePlatform.imageID` still tells "_For **compute machines**, the image ID that..._", while in fact the parameter is for **both compute machines and control plane machines**, if not otherwise specified. 
2. I also checked [aws doc](https://docs.openshift.com/container-platform/4.11/installing/installing_aws/installing-aws-vpc.html#installation-configuration-parameters-optional-aws_installing-aws-vpc) (table 4) and [azure doc](https://docs.openshift.com/container-platform/4.11/installing/installing_azure/installing-azure-vnet.html#installation-configuration-parameters-additional-azure_installing-azure-vnet) (table 5) on the similar table content, although neither is perfect, both firstly list compute parameters (e.g. `compute.platform.aws.amiID` in aws doc), then controlPlane parameters (e.g. `controlPlane.platform.aws.amiID` in aws doc), and then defaultMachinePlatform parameters (e.g. `platform.aws.defaultMachinePlatform.amiID` in aws doc). 
3. In order to be consistent with other provider doc, suggest to also firstly list `compute.platform.alibabacloud` parameters, then `controlPlane.platform.alibabacloud` parameters, and then `platform.alibabacloud.defaultMachinePlatform` parameters (along with other `platform.alibabacloud` parameters) in table 4, and explicitly tell in each parameter's Description field that the parameter is for compute machines, or control plane machines, or both. I think this way would be much clear, HDYT? 
4. If you agree, please remove the statements/notes added last time.

GroceryBoyJr commented:
No bother at all @jianli-wei and @twiest I am always happy to have your feedback. i will address and notify you when done!

GroceryBoyJr commented:
> @GroceryBoyJr Sorry, it seems I didn't put it very clear on the table 4 suggestions, let me clarify. Very sorry to bother you again.

Thank you so much, appreciate the clarification, no bother at all!

> 1. Now in table 4, e.g. the Description field of `platform.alibabacloud.defaultMachinePlatform.imageID` still tells "_For **compute machines**, the image ID that..._", while in fact the parameter is for **both compute machines and control plane machines**, if not otherwise specified.

I have added this individually for each option. 

> 2. I also checked [aws doc](https://docs.openshift.com/container-platform/4.11/installing/installing_aws/installing-aws-vpc.html#installation-configuration-parameters-optional-aws_installing-aws-vpc) (table 4) and [azure doc](https://docs.openshift.com/container-platform/4.11/installing/installing_azure/installing-azure-vnet.html#installation-configuration-parameters-additional-azure_installing-azure-vnet) (table 5) on the similar table content, although neither is perfect, both firstly list compute parameters (e.g. `compute.platform.aws.amiID` in aws doc), then controlPlane parameters (e.g. `controlPlane.platform.aws.amiID` in aws doc), and then defaultMachinePlatform parameters (e.g. `platform.aws.defaultMachinePlatform.amiID` in aws doc).

OK, I see that also.

> 3. In order to be consistent with other provider doc, suggest to also firstly list `compute.platform.alibabacloud` parameters, then `controlPlane.platform.alibabacloud` parameters, and then `platform.alibabacloud.defaultMachinePlatform` parameters (along with other `platform.alibabacloud` parameters) in table 4, and explicitly tell in each parameter's Description field that the parameter is for compute machines, or control plane machines, or both. I think this way would be much clear, HDYT?

I do not have any compute.platform.alibabacloud or controlPlane.platform.alibabacloud optional parameters in Table 4. All I have are platform.alibabacloud and platform.alibabacloud.defaultMachinePlatform. I don't understand?

If there are missing values, is it possible, please, for me to move ahead with this PR then open a new one to correct the table only?

> 4. If you agree, please remove the statements/notes added last time.
I have removed or edited the statements. I kept the form of the text before the table because it is clear, but I didn't completely delete, wanted to emphasize to the customer some parameters apply to both compute and control plane.

GroceryBoyJr commented:
@twiest @jianli-wei have made changes to Table 4 and pushed them to the preview.

jianli-wei commented:
> I do not have any compute.platform.alibabacloud or controlPlane.platform.alibabacloud optional parameters in Table 4. All I have are platform.alibabacloud and platform.alibabacloud.defaultMachinePlatform. I don't understand?

@GroceryBoyJr My bad again, I'd thought you know the installer's `explain` command. For example, start with `openshift-install explain installconfig` and then dig into each interested field of "object" (or list of "object") type. 
```
$ openshift-install version
openshift-install 4.12.0-0.nightly-2022-09-28-204419
built from commit 9eb0224926982cdd6cae53b872326292133e532d
release image registry.ci.openshift.org/ocp/release@sha256:2c8e617830f84ac1ee1bfcc3581010dec4ae5d9cad7a54271574e8d91ef5ecbc
release architecture amd64
$ 
$ openshift-install explain installconfig.platform.alibabacloud.defaultMachinePlatform
KIND:     InstallConfig
VERSION:  v1

RESOURCE: <object>
  DefaultMachinePlatform is the default configuration used when installing on Alibaba Cloud for machine pools which do not define their own platform configuration.

FIELDS:
    imageID <string>
      ImageID is the Image ID that should be used to create ECS instance. If set, the ImageID should belong to the same region as the cluster.

    instanceType <string>
      InstanceType defines the ECS instance type. eg. ecs.g6.large

    systemDiskCategory <string>
      Valid Values: "","cloud_efficiency","cloud_essd"
      SystemDiskCategory defines the category of the system disk.

    systemDiskSize <integer>
      SystemDiskSize defines the size of the system disk in gibibytes (GiB).

    zones <[]string>
      Zones is list of availability zones that can be used. eg. ["cn-hangzhou-i", "cn-hangzhou-h", "cn-hangzhou-j"]

$ 
$ openshift-install explain installconfig.compute.platform.alibabacloud
KIND:     InstallConfig
VERSION:  v1

RESOURCE: <object>
  AlibabaCloud is the configuration used when installing on Alibaba Cloud.

FIELDS:
    imageID <string>
      ImageID is the Image ID that should be used to create ECS instance. If set, the ImageID should belong to the same region as the cluster.

    instanceType <string>
      InstanceType defines the ECS instance type. eg. ecs.g6.large

    systemDiskCategory <string>
      Valid Values: "","cloud_efficiency","cloud_essd"
      SystemDiskCategory defines the category of the system disk.

    systemDiskSize <integer>
      SystemDiskSize defines the size of the system disk in gibibytes (GiB).

    zones <[]string>
      Zones is list of availability zones that can be used. eg. ["cn-hangzhou-i", "cn-hangzhou-h", "cn-hangzhou-j"]

$ 
$ openshift-install explain installconfig.controlPlane.platform.alibabacloud
KIND:     InstallConfig
VERSION:  v1

RESOURCE: <object>
  AlibabaCloud is the configuration used when installing on Alibaba Cloud.

FIELDS:
    imageID <string>
      ImageID is the Image ID that should be used to create ECS instance. If set, the ImageID should belong to the same region as the cluster.

    instanceType <string>
      InstanceType defines the ECS instance type. eg. ecs.g6.large

    systemDiskCategory <string>
      Valid Values: "","cloud_efficiency","cloud_essd"
      SystemDiskCategory defines the category of the system disk.

    systemDiskSize <integer>
      SystemDiskSize defines the size of the system disk in gibibytes (GiB).

    zones <[]string>
      Zones is list of availability zones that can be used. eg. ["cn-hangzhou-i", "cn-hangzhou-h", "cn-hangzhou-j"]

$

jianli-wei commented on modules/installation-configuration-parameters.adoc:
```diff
@@ -1544,23 +1557,23 @@ Additional Alibaba Cloud configuration parameters are described in the following
 
 |`platform.alibabacloud.defaultMachinePlatform.instanceType`
-|For compute machines, the configuration used when installing on Alibaba Cloud.
+|For both compute machines and control plane machines, the configuration used when installing on Alibaba Cloud.
```
The statement doesn't tell the exact meaning of the parameter, suggest to replace `the configuration used when installing on Alibaba Cloud` with `the ECS instance type that should be used to create ECS instance`.

jianli-wei commented on modules/installation-configuration-parameters.adoc:
```diff
@@ -1544,23 +1557,23 @@ Additional Alibaba Cloud configuration parameters are described in the following
 |`platform.alibabacloud.defaultMachinePlatform.zones`
-|For compute machine, list of availability zones that can be used.
+|For both compute machines and control plane machines, list of availability zones that can be used.
 |String.
```
It should be `String list`.

GroceryBoyJr commented:
@jianli-wei You are a hard man sir, but I will comply. :-D :-D :-D

GroceryBoyJr commented:
@jianli-wei @twiest  Do you know if there is a 120 GiB minimum for controlPlane.platform.alibabacloud.systemDisksize ?

GroceryBoyJr commented:
@jianli-wei @twiest all requested updates have been made - please review at the preview link at this time. I welcome any additions or changes you request for the document.

jianli-wei commented on modules/installation-configuration-parameters.adoc:
```diff
@@ -1516,13 +1520,66 @@ ifdef::alibabacloud[]
+
+|`compute.platform.alibabacloud.zones`
+|The list of availability zones that can be used. Examples: `cn-hangzhou-h`, `cn-hangzhou-j`
+|String.
```
It should be "String list", see line 1577.

jianli-wei commented on modules/installation-configuration-parameters.adoc:
```diff
@@ -1516,13 +1520,66 @@ ifdef::alibabacloud[]
+
+[NOTE]
+====
+If defined, the parameters `compute.platform.alibabacloud` and `controlPlane.platform.alibabacloud` will overwrite `platform.alibabacloud.defaultMachinePlatform` for compute machines and control plane machines respectively.
```
The suggestion: 
If defined, the parameters under `compute.platform.alibabacloud` and `controlPlane.platform.alibabacloud` will overwrite `platform.alibabacloud.defaultMachinePlatform` settings for compute machines and control plane machines respectively.

GroceryBoyJr commented:
@twiest Tom - you still have a "Changes Requested" showing on this PR 44593. Check me please sir, is there anything you asked for that I have missed?

jianli-wei commented on modules/installation-configuration-parameters.adoc:
```diff
@@ -1516,13 +1520,66 @@ ifdef::alibabacloud[]
+|The list of availability zones that can be used. Examples: `cn-hangzhou-h`, `cn-hangzhou-j`
+|String list.
+
+|`platform.alibabacloud.privateZoneID`
```
Did you notice the parameter `platform.alibabacloud.privateZoneID` was shown twice in the table 4? It's shown in both line 1579 and line 1623, please correct it, thanks!

jianli-wei commented on modules/installation-configuration-parameters.adoc:
```diff
@@ -1516,13 +1520,66 @@ ifdef::alibabacloud[]
+|The list of availability zones that can be used. Examples: `cn-hangzhou-h`, `cn-hangzhou-j`
+|String list.
+
+|`platform.alibabacloud.privateZoneID`
```
Did you notice the parameter `platform.alibabacloud.privateZoneID` was shown twice in the table 4? It's shown in both line 1579 and line 1623, please correct it, thanks!

jianli-wei replied:
@GroceryBoyJr Any updates on this? Thanks!

twiest reviewed (approved):
LGTM

twiest commented:
> @jianli-wei @twiest Do you know if there is a 120 GiB minimum for controlPlane.platform.alibabacloud.systemDisksize ?

I'm not aware of this requirement. Where are you seeing it?

jianli-wei commented:
> > @jianli-wei @twiest Do you know if there is a 120 GiB minimum for controlPlane.platform.alibabacloud.systemDisksize ?
> 
> I'm not aware of this requirement. Where are you seeing it?

@GroceryBoyJr FYI https://docs.openshift.com/container-platform/4.11/installing/installing_gcp/installing-gcp-customizations.html#installation-minimum-resource-requirements_installing-gcp-customizations

GroceryBoyJr commented:
@twiest and @jianli-wei, PTAL?

GroceryBoyJr commented:
/label peer-review-requested

GroceryBoyJr commented:
/label peer-review-needed

jianli-wei commented:
> @twiest and @jianli-wei, PTAL? Preview is at: http://file.rdu.redhat.com/jbrigman/alibaba-vpc/installing/installing_alibaba/installing-alibaba-vpc.html

@GroceryBoyJr I reviewed it, one point, Table 4 tells `platform.alibabacloud.privateZoneID` twice, so please remove one, thanks!

FYI it's said Thomas Wiest is no longer with us in Red Hat.

Amrita42 commented:
/label peer-review-in-progress

Amrita42 commented:
@GroceryBoyJr please add a bz prefix to the title . Also in the description , please mention the full bz link.

Amrita42 commented on installing/installing_alibaba/installing-alibaba-vpc.adoc:
```diff
@@ -0,0 +1,71 @@
+
+toc::[]
+
+In {product-title} version {product-version}, you can install a cluster into an existing Alibaba Virtual Private Cloud (VPC) on Alibaba Cloud Services. The installation program provisions the required infrastructure which can then be customized. To customize the VPC installation, modify parameters in the 'install-config.yaml' file before you install the cluster.
```
```suggestion
In {product-title} version {product-version}, you can install a cluster into an existing Alibaba Virtual Private Cloud (VPC) on Alibaba Cloud Services. The installation program provisions the required infrastructure, which can then be customized. To customize the VPC installation, modify the parameters in the 'install-config.yaml' file before you install the cluster.
```

GroceryBoyJr replied:
I had to re-do this edit in the restored files of my new PR. Thank you so much for your detailed review, @Amrita42 !

Amrita42 commented on modules/installation-alibaba-config-yaml.adoc:
```diff
@@ -48,14 +49,24 @@ platform:
+<5> Optional, Example vpcID value.
+<6> Optional, Example vswitchID values.
+<7> Required. The installation program prompts you for the pull secret.
+<8> Optional. The installation program prompts you for the SSH key value that you use to access the machines in your cluster.
```
```suggestion
<8> Optional. The installation program prompts you for the SSH key value that you will use to access the machines in your cluster.
```

GroceryBoyJr replied:
Done! Thank you @Amrita42 !

Amrita42 commented on modules/installation-custom-alibaba-vpc.adoc:
```diff
@@ -0,0 +1,52 @@
+[id="installation-custom-alibaba-vpc_{context}"]
+= About using a custom VPC
+
+In {product-title} {product-version}, you can deploy a cluster into existing subnets in an existing Virtual Private Cloud (VPC) in Alibaba Cloud Platform. By deploying {product-title} into an existing Alibaba VPC, you might be able to avoid limit constraints in new accounts or more easily abide by the operational constraints that your company's guidelines set. If you cannot obtain the infrastruct ...
```
```suggestion
In {product-title} {product-version}, you can deploy a cluster into existing subnets in an existing Virtual Private Cloud (VPC) in the Alibaba Cloud Platform. By deploying {product-title} into an existing Alibaba VPC, you might be able to avoid limit constraints in new accounts or more easily abide by the operational constraints that your company's guidelines set. If you cannot obtain the infrastructure creation permissions that are required to create the VPC yourself, use this installation option. You must configure networking using vSwitches.
```
```suggestion
In {product-title} {product-version}, you can deploy a cluster into existing subnets in an existing Virtual Private Cloud (VPC) in Alibaba Cloud Platform. By deploying {product-title} into an existing Alibaba VPC, you might be able to avoid limit constraints in new accounts or more easily abide by the operational constraints that your company's guidelines set. If you cannot obtain the infrastructure creation permissions that are required to create the VPC yourself, use this installation option. You must configure networking using vSwitches.
```
```suggestion
In {product-title} {product-version}, you can deploy a cluster into existing subnets in an existing Virtual Private Cloud (VPC) in the Alibaba Cloud Platform. By deploying {product-title} into an existing Alibaba VPC, you might be able to avoid limit constraints in new accounts or more easily abide by the operational constraints that adhere to your company's guidelines. If you cannot obtain the infrastructure creation permissions that are required to create the VPC yourself, use this installation option. You must configure networking using vSwitches.
```

Amrita42 replied:
oops , please look at the last suggestion :)

GroceryBoyJr replied:
OK! Got it, all good, thank you!

Amrita42 commented on modules/installation-custom-alibaba-vpc.adoc:
```diff
@@ -0,0 +1,52 @@
+To ensure that the vSwitches you provide are suitable, the installation program confirms the following data:
+
+* All the vSwitches that you specify must exist.
+* You provide one or more vSwitches for control plane machines and compute machines.
```
```suggestion
* You have provided one or more vSwitches for control plane machines and compute machines.
```

GroceryBoyJr replied:
Done, thank you!

Amrita42 commented on modules/installation-custom-alibaba-vpc.adoc:
```diff
@@ -0,0 +1,52 @@
+[id="installation-about-custom-alibaba-permissions_{context}"]
+== Division of permissions
+
+Some individuals can create different resources in your cloud than others. For example, you might be able to create application-specific items, like instances, buckets, and load balancers, but not networking-related components such as VPCs or vSwitches.
```
```suggestion
Some individuals can create different resources in your cloud than others. For example, you might be able to create application-specific items, like instances, buckets, and load balancers, but not networking-related components, such as VPCs or vSwitches.
```

GroceryBoyJr replied:
Done, thank you!

Amrita42 commented on modules/installation-custom-alibaba-vpc.adoc:
```diff
@@ -0,0 +1,52 @@
+[id="installation-custom-alibaba-vpc-isolation_{context}"]
+== Isolation between clusters
+
+If you deploy {product-title} to an existing network, the isolation of cluster services is reduced in the following ways:
```
I am not sure , is t_o _to an_ existing network or into an existing network_.

GroceryBoyJr replied:
I'll accept your suggested change. It seems like it might aid localization.

Amrita42 commented on modules/installation-custom-alibaba-vpc.adoc:
```diff
@@ -0,0 +1,52 @@
+
+* You can install multiple {product-title} clusters in the same VPC.
+
+* ICMP ingress is allowed to the entire network.
```
into vs to the network

GroceryBoyJr replied:
I'm going to leave that one as is, because I believe it is "access to" and not "access into". In this case ICMP ingress is an access permission to a network, not access permission into a network. It's different than the usage above because this is not deployment to, it is a recurring access to.

Amrita42 commented:
/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

GroceryBoyJr commented:
Reopen if possible

GroceryBoyJr commented:
> @GroceryBoyJr please add a bz prefix to the title . Also in the description , please mention the full bz link.

Amrita - there is no BZ for this, there is only the JIRA
